### PR TITLE
Bug 1142546 - HomePanelDelegate needs to be able to optionally open pages in a new tab

### PR DESCRIPTION
**Pull request by Johennes**

### CODE REVIEW

File: Client/Frontend/Home/ReaderPanel.swift [hunk 1]
```diff
@@ -442,7 +442,7 @@ class ReadingListPanel: UITableViewController, HomePanel, SWTableViewCellDelegat
             profile.readingList?.updateRecord(record, unread: false)
             // Reading list items are closest in concept to bookmarks.
             let visitType = VisitType.Bookmark
-            homePanelDelegate?.homePanel(self, didSelectURL: encodedURL, visitType: visitType)
+            homePanelDelegate?.homePanel(self, didSelectURL: encodedURL, visitType: visitType, inNewTab: true)
```
Comment (Johennes) [inline Client/Frontend/Home/ReaderPanel.swift]: I made links from the reader panel open in new tabs. All others open in the current tab.

File: Client/Frontend/Home/HomePanelViewController.swift [hunk 2]
```diff
@@ -17,7 +17,7 @@ private struct HomePanelViewControllerUX {
 }
 
 protocol HomePanelViewControllerDelegate: class {
-    func homePanelViewController(homePanelViewController: HomePanelViewController, didSelectURL url: NSURL, visitType: VisitType)
+    func homePanelViewController(homePanelViewController: HomePanelViewController, didSelectURL url: NSURL, visitType: VisitType, inNewTab: Bool)
```
Comment (Johennes) [inline Client/Frontend/Home/HomePanelViewController.swift]: Unfortunately, Swift doesn't allow default parameters in protocol methods so I had to adapt all callers of this method.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -829,6 +829,7 @@ class BrowserViewController: UIViewController {
     }
 
     func openURLInNewTab(url: NSURL) {
+        urlBar.leaveOverlayMode()
```
Comment (Johennes) [inline Client/Frontend/Browser/BrowserViewController.swift]: I had to add this to get back from the home panel to the browser after opening a link in a new tab.

**Comment by sleroux** [comment]
@Johennes This is a neat patch. Seems low risk and adds the benefit of opening reading list links in new tabs. In Bugzilla it was mentioned we should wait until we have a need for this but I don't see the harm in landing this. @st3fan thoughts?

**Comment by Johennes** [comment]
@sleroux / @st3fan: Let me know in case you indeed decide to land this and I'll rebase it.

**Comment by farhanpatel** [comment]
I'm closing this because this is very old now. Also because this is landing in Activity Stream itself which will replace the home panels.

**Pull request closed without merging**